Spigot-API-Patches/0237-add-recipe-to-cook-events.patch

```diff
@@ -0,0 +1,76 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Thonk <30448663+ExcessiveAmountsOfZombies@users.noreply.github.com>
+Date: Thu, 19 Nov 2020 18:30:03 -0600
+Subject: [PATCH] add recipe to cook events
+
+
+diff --git a/src/main/java/org/bukkit/event/block/BlockCookEvent.java b/src/main/java/org/bukkit/event/block/BlockCookEvent.java
+index be7af5440bf9923f0c9c84efa4d70a89337a2f96..89c97e1205598174b4751d0e194303add4420a26 100644
+--- a/src/main/java/org/bukkit/event/block/BlockCookEvent.java
++++ b/src/main/java/org/bukkit/event/block/BlockCookEvent.java
+@@ -4,7 +4,9 @@ import org.bukkit.block.Block;
+ import org.bukkit.event.Cancellable;
+ import org.bukkit.event.HandlerList;
+ import org.bukkit.inventory.ItemStack;
++import org.bukkit.inventory.Recipe; // Paper
+ import org.jetbrains.annotations.NotNull;
++import org.jetbrains.annotations.Nullable; // Paper
+ 
+ /**
+  * Called when an ItemStack is successfully cooked in a block.
+@@ -12,15 +14,30 @@ import org.jetbrains.annotations.NotNull;
+ public class BlockCookEvent extends BlockEvent implements Cancellable {
+     private static final HandlerList handlers = new HandlerList();
+     private final ItemStack source;
++    private final Recipe recipe; // Paper
+     private ItemStack result;
+     private boolean cancelled;
+ 
+     public BlockCookEvent(@NotNull final Block block, @NotNull final ItemStack source, @NotNull final ItemStack result) {
++        this(block, source, result, null); // Paper
++    }
++
++    // Paper start
++    public BlockCookEvent(@NotNull final Block block, @NotNull final ItemStack source, @NotNull final ItemStack result, @Nullable Recipe recipe) {
+         super(block);
+         this.source = source;
+         this.result = result;
+         this.cancelled = false;
++        this.recipe = recipe;
++    }
++    // Paper end
++
++    // Paper start
++    @Nullable
++    public Recipe getRecipe() {
++        return recipe;
+     }
++    // Paper end
```

Comment:
The above Paper - start/end section can probably be combined with the one above it. Kinda redundant to have a `// Paper end` and `// Paper start` comment right after it.